datafusion/src/datasource/object_store/mod.rs

```diff
@@ -141,7 +141,7 @@ pub trait ObjectStore: Sync + Send + Debug {
         let suffix = suffix.to_owned();
         Ok(Box::pin(file_stream.filter(move |fr| {
             let has_suffix = match fr {
-                Ok(f) => f.path().ends_with(&suffix),
+                Ok(f) => do_path_filter_with_suffix(f.path(), &suffix),
```

Comment:
Forcing the filter here will restrict the usefullness of the object store abstraction. for example, I won't be able to use `list_file_with_suffix` in delta-rs to manage delta lake metadata. Object store should be a very generic module that is agnostic of this type of application specific file naming conventions. I think a better place to host this particular logic is in the ListingTable struct.

Reply:
@houqp  thanks a lot for your kindly explain!  
i got that object_store should be highly abstract. This filter login from spark is in `HadoopFSUtils`. So i think maybe i will keep the code clean, put this in out customized ObjectStore for hadoop.